### PR TITLE
Fix/clusterfile not exist

### PR DESCRIPTION
**Pull request by bxy4543**
optimize errors caused by clusterfile not found.

add Cluster check

### CODE REVIEW

File: apply/default_applier.go [hunk 1]
```diff
@@ -120,8 +121,33 @@ var ActionFuncMap = map[ActionName]func(*DefaultApplier) error{
 	CleanFS: func(applier *DefaultApplier) error {
 		return applier.FileSystem.Clean(applier.ClusterDesired)
 	},
+	CheckDesiredCluster: func(applier *DefaultApplier) error {
+		return checkDesiredCluster(applier)
+	},
 }
 
+func checkDesiredCluster(applier *DefaultApplier) error {
```
Comment (fanux) [inline apply/default_applier.go:129]: I think we can add this function to the check module.

Comment (fanux) [reply apply/default_applier.go:129]: pre check

File: utils/hosts.go [hunk 2]
```diff
@@ -38,3 +38,27 @@ func GetDiffHosts(hostsOld, hostsNew v1.Hosts) (add, sub []string) {
 
 	return
 }
+
+func HasSameHosts(hostsOld, hostsNew v1.Hosts) bool {
```
Comment (kakaZhou719) [inline utils/hosts.go:42]: i think only IPlist we need in this function. (hostsOld, hostsNew []sting)

File: check/checker/types.go [hunk 3]
```diff
@@ -19,3 +19,8 @@ import v1 "github.com/alibaba/sealer/types/api/v1"
 type Checker interface {
 	Check(cluster *v1.Cluster) error
 }
+
+// PreChecker: checking prerequisite for sealer,such as system info,register login status.
+type PreChecker interface {
```
Comment (fanux) [inline check/checker/types.go:24]: I think we don't need this interface, just using Checker interface

**Review by fanux — APPROVED** [review]

**Pull request closed without merging**